uc_store/classes/mail.inc

```diff
@@ -23,4 +23,19 @@ class UbercartMailSystem extends DefaultMailSystem {
     return $message;
   }
 
+  /**
+   * Implements MailSystemInterface::mail().
+   *
+   * Retrieve the configured class responsible for delivering a message
+   * defined by system.mail. Forwards the call to that class.
+   */
+   public function mail(array $message) {
+    $mailsystem_name = config_get('system.mail', 'default-system');
+    // Don't allow circular references; fall back to the core default instead.
+    if ($mailsystem_name == 'UbercartMailSystem') {
```

Comment:
One more check to be on the safe side re: circular references

```suggestion
    if ($mailsystem_name == 'UbercartMailSystem' || is_subclass_of($mailsystem_name, 'UbercartMailSystem') {
```

This is to prevent the situation when UbercartMailSystem is extended and the extension is declared in `system.mail` as the default class, AND the new class does not override `mail()`. In which case, the original `UbercartMailSystem::mail()` would be used, creating a circular reference, and the method creating a new instance of itself and calling `mail()` ad infinitum. If the extending class overrides `mail()`, then `UbercartMailSystem::mail()` would not be run.